sunshine/stream.cpp

```diff
@@ -763,7 +763,7 @@ void controlBroadcastThread(control_server_t *server) {
       break;
     }
 
-    server->iterate(50ms);
+    server->iterate(500ms);
```

Comment:
It checks every 50ms if it should send a rumble event

Reply:
Alright, so it won't be *that* easy then :D

Reply:
Does it affect your CPU usage?
The code for sending/receiving requires synchronisation, so if it doesn't affect the CPU usage, it's not really worth it in my opinion to increase the duration of sleep.

:)

Reply:
Yeah it does. Capture thread was around 35% and this (control thread) 15%.

Reply:
I'll have to think on how to best implement it. 🤔

Reply:
In the branch `lower-cpu-usage-controlthread`, I've modified the rumble code to be based on callbacks rather than a queue.
I've also merged your changes there.

Unfortunately, it's using locks for synchronization due to a limitation in the library `Enet`.
Could you compare the branch `lower-cpu-usage-controlthread` with your own branch with the following modification?:
`server->iterate(500ms);` --> `server->iterate(150ms);`

Reply:
Looks good also with 150ms, the cpu usage is slightly higher than 500ms of course, but still *much* lower than what it was before.